tools/codegen/gen_functionalization_type.py

```diff
@@ -0,0 +1,426 @@
+        outputs = []
+        for f in fs:
+            if not is_view_op(f) and not modifies_arguments(f):
+                continue
+            if self.target is Target.REGISTRATION:
+                output = self.emit_registration(f)
+            elif self.target is Target.DEFINITION:
+                if is_view_op(f) and f.has_composite_implicit_autograd_kernel:
+                    # See Note [Composite view ops in the functionalization pass]
+                    continue
+                output = self.emit_definition(f, group)
+            else:
+                assert_never(self.target)
+
+            outputs.append(output)
+        return outputs
+
+    def emit_registration(self, f: NativeFunction) -> str:
+        if is_view_op(f) and f.has_composite_implicit_autograd_kernel:
+            assert self.composite_implicit_autograd_index is not None
+            metadata = self.composite_implicit_autograd_index.get_kernel(f)
+            assert metadata is not None
+            native_api_name = metadata.kernel
+            sig = DispatcherSignature.from_schema(f.func)
+            # Note [Composite view ops in the functionalization pass]
+            # We don't need to worry about implemententing functionalization kernels for views with
+            # CompositeImplicitAutograd kernels, because we can just decompose them into their base operators.
+            # We can't just opt the entire Functionalization dispatch key into the composite keyset though,
+            # because we don't want to decompose non-view ops that are composite, like `at::ones`.
+            registration_str = f'static_cast<{sig.ptr_type()}>(at::native::{native_api_name})'
+        else:
+            registration_str = f'TORCH_FN(functionalization::{wrapper_name(f.func)})'
+
+        return f"""
+    m.impl("{f.func.name}",
+        {registration_str}
+    );
+    """
+
+    def emit_definition(self, f: NativeFunction, g: Optional[NativeFunctionsGroup]) -> str:
+        # order is important here, ops that are both views and mutations should hit the view path.
+        if is_view_op(f):
+            body_str = emit_view_functionalization_body(f, g)
+        else:
+            # inplace op
+            assert modifies_arguments(f)
+            body_str = emit_inplace_functionalization_body(f, g)
+        sig = DispatcherSignature.from_schema(f.func)
+        return f"""
+    {sig.defn(name=wrapper_name(f.func), is_redispatching_fn=True)} {{
+    {body_str}
+    }}
+    """
+
+# See Note [Functionalization Pass: View Inverses].
+def gen_functionalization_view_inverse_declaration(g: Union[NativeFunction, NativeFunctionsGroup]) -> List[str]:
+    fs = [g] if isinstance(g, NativeFunction) else g.functions()
+    outputs = []
+    for f in fs:
+        if is_view_op(f) and not f.has_composite_implicit_autograd_kernel:
+            output = emit_declaration_for_noncomposite_views(f)
+            outputs.append(output)
+    return outputs
```

Comment:
unimportant fyi (don't make a code change): mapMaybe would work here; you'd return None if the condition was false.